roxie/ccd/ccdserver.cpp

```diff
@@ -17033,8 +17051,23 @@ class CRoxieServerNWayBaseActivity : public CRoxieServerMultiInputBaseActivity
 
     virtual void stop()
     {
-        ForEachItemIn(i, expandedStreams)
-            expandedStreams.item(i)->stop();
+        for (unsigned i=0; i < numInputs; i++)
+        {
+            IFinalRoxieInput * cur = inputArray[i];
+            CRoxieServerNWayInputBaseActivity *nWayInput = dynamic_cast<CRoxieServerNWayInputBaseActivity *>(cur);
+            if (nWayInput)
+            {
+                unsigned numRealInputs = cur->numConcreteOutputs();
```

Comment:
This also will not have been set up if the activity hasn't been started.  I'm not sure it improves the situation at all.